src/stochastics/fractional_processes.rs

```diff
@@ -0,0 +1,255 @@
+    /// Vector of process trajectories.
+    pub paths: Vec<Vec<f64>>,
+}
+
+/// Trait to implement stochastic processes.
+pub trait FractionalStochasticProcess: Sync {
+    /// Base method for the process' drift.
+    fn drift(&self, x: f64, t: f64) -> f64;
+
+    /// Base method for the process' diffusion.
+    fn diffusion(&self, x: f64, t: f64) -> f64;
+
+    /// Base method for the process' jump term (if applicable).
+    fn jump(&self, x: f64, t: f64) -> f64;
+
+    /// Autocovariance function.
+    fn afc_vector(&self, n: usize, hurst: f64) -> RowDVector<f64> {
+        let mut v = RowDVector::<f64>::zeros(n);
+        v[0] = 1.0;
+
+        for i in 1..n {
+            let idx = i as f64;
+
+            v[i] = 0.5
+                * ((idx + 1.0).powf(2.0 * hurst) - 2.0 * idx.powf(2.0 * hurst)
+                    + (idx - 1.0).powf(2.0 * hurst))
+        }
+
+        v
+    }
+
+    /// Autocovariance matrix.
+    fn afc_matrix_sqrt(&self, n: usize, hurst: f64) -> DMatrix<f64> {
+        let acf_v = self.afc_vector(n, hurst);
+        let mut m = DMatrix::<f64>::zeros_generic(Dyn::from_usize(n), Dyn::from_usize(n));
+
+        for i in 0..n {
+            for j in 0..n {
+                match i.cmp(&j) {
+                    Equal => m[(i, j)] = acf_v[0],
+                    Greater => m[(i, j)] = acf_v[i - j],
+                    Less => continue,
+                }
+            }
+        }
+
+        m.cholesky().unwrap().l()
+    }
+
+    /// Fractional Gaussian noise.
+    fn fgn(&self, n: usize, hurst: f64) -> RowDVector<f64> {
+        let acf_sqrt = self.afc_matrix_sqrt(n, hurst);
+        let noise = rand::thread_rng()
+            .sample_iter::<f64, StandardNormal>(StandardNormal)
+            .take(n)
+            .collect();
+        let noise = DVector::<f64>::from_vec(noise);
+
+        (acf_sqrt * noise).transpose() * (n as f64).powf(-hurst)
+    }
+
+    /// Fractional Gaussian noise.
+    #[cfg(feature = "seedable")]
+    fn seedable_fgn(&self, n: usize, hurst: f64, seed: u64) -> RowDVector<f64> {
+        use rand::rngs::StdRng;
+
+        let acf_sqrt = self.afc_matrix_sqrt(n, hurst);
+        let noise = StdRng::seed_from_u64(seed)
+            .sample_iter::<f64, StandardNormal>(StandardNormal)
+            .take(n)
+            .collect();
+        let noise = DVector::<f64>::from_vec(noise);
+
+        (acf_sqrt * noise).transpose() * (n as f64).powf(-hurst)
+    }
+
+    /// Euler-Maruyama discretisation scheme.
+    ///
+    /// # Arguments:
+    /// * `x_0` - The process' initial value at `t_0`.
+    /// * `t_0` - The initial time point.
+    /// * `t_n` - The terminal time point.
+    /// * `n_steps` - The number of time steps between `t_0` and `t_n`.
+    /// * `m_paths` - How many process trajectories to simulate.
+    /// * `parallel` - Run in parallel or not (recommended for > 1000 paths).
+    /// * `hurst` - The Hurst parameter of the fractional Brownian motion.
+    fn euler_maruyama(
```

Comment:
My first version was as you write, I tried to pass the hurst exponent into the `process.rs` but I think it was a bit messy. The process with fractional noise is not as obvious as the ordinary one, so I thought to build a full standalone fractional stochastic process part of the package but with the same methodology that the package follows. In the future, I want to add a more efficient method based on the Davies-Harte family with circulant embedding matrix and FFT so that was the reason why I made that.

Reply:
If we use the Euler method for generating sequences it is possible to build the fractional stochastic analysis part in the same way as the ordinary. I can merge it into the `process.rs` but I need the ACF methods to because I need to generate a fractional Brownian motion as the random noise to certain processes. What do you think, which one fits better for your needs? :)